src/auth/auth.service.ts

```diff
@@ -25,48 +17,41 @@ export class AuthService {
     private configService: ConfigService,
   ) {}
 
-  async login(loginDto: LoginDto, response: Response) {
-    const dbUser = await this.userService.getUser(loginDto.username);
-    if (!dbUser) {
-      Logger.error('User not found');
-      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
-    }
-    const isPasswordMatching = await bcrypt.compare(
-      loginDto.password,
-      dbUser.password,
-    );
-    if (!isPasswordMatching) {
-      Logger.error('password not matching');
-      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
-    }
+  async login(user: User): Promise<{ access_token: string }> {
     const tokenPayload: TokenPayload = {
-      sub: dbUser.id,
-      username: dbUser.username,
+      sub: user.id,
+      username: user.username,
+      roles: user.roles,
     };
-    const jwtToken = this.jwtService.sign(tokenPayload);
+    const jwtToken = await this.jwtService.signAsync(tokenPayload);
 
     const expires = new Date();
     expires.setSeconds(
       expires.getSeconds() + this.configService.get('JWT_EXPIRATION'),
     );
-    response.cookie('Authentication', jwtToken, {
-      httpOnly: true,
-      expires: TWO_HOURS_FROM_NOW_DATE,
-    });
-    return jwtToken;
+
+    return { access_token: jwtToken };
   }
 
-  async register(createUserDto: CreateUserDto): Promise<User> {
+  async register(
+    createUserDto: CreateUserDto,
+    roles: Role[] = [Role.USER],
+  ): Promise<User> {
     try {
-      const isUserExist =
-        (await this.userService.getUser(createUserDto.username)) !== null;
+      const isUserExist = await this.userService.isUserExistsCheck(
+        createUserDto.username,
+        createUserDto.email,
+      );
       if (isUserExist) {
         throw new HttpException(
           ERROR_MESSAGES.USER_ALREADY_EXISTS,
           HttpStatus.BAD_REQUEST,
```

Comment:
https://docs.nestjs.com/exception-filters

Nest has built in HTTP Exceptions why not use them?

Reply:
it is nests HttpExeption my friend